controllers/virtinkmachine_controller.go

```diff
@@ -100,39 +151,33 @@ func (r *VirTinkMachineReconciler) reconcile(ctx context.Context, machine *infra
 		Namespace: machine.Namespace,
 	}
 	vmNotFound := false
-	if err := r.Get(ctx, vmKey, &vm); err != nil {
+	if err := externalCluster.Get(ctx, vmKey, &vm); err != nil {
 		if apierrors.IsNotFound(err) {
 			vmNotFound = true
 		} else {
-			return fmt.Errorf("get VM: %s", err)
+			return ctrl.Result{}, fmt.Errorf("get VM from external Cluster: %s", err)
 		}
 	}
 
-	if !vmNotFound && !metav1.IsControlledBy(&vm, machine) {
-		vmNotFound = true
-	}
-
 	if vmNotFound {
 		vm, err := r.buildVM(ctx, machine, ownerMachine)
 		if err != nil {
-			return fmt.Errorf("build VM: %s", err)
+			return ctrl.Result{}, fmt.Errorf("build VM: %s", err)
 		}
 
 		vm.Name = vmKey.Name
 		vm.Namespace = vmKey.Namespace
-		if err := controllerutil.SetControllerReference(machine, vm, r.Scheme); err != nil {
-			return fmt.Errorf("set VM controller reference: %s", err)
-		}
-		if err := r.Create(ctx, vm); err != nil {
-			return fmt.Errorf("create VM: %s", err)
+		if err := externalCluster.Create(ctx, vm); err != nil {
+			return ctrl.Result{}, fmt.Errorf("create VM in external Cluster: %s", err)
 		}
 		r.Recorder.Eventf(machine, corev1.EventTypeNormal, "CreatedVM", "Create VM %q", vm.Name)
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Add a `ctrl.Result` return var is an overkill to just requeue the object. Could we return a "requeue error" here and capture it in the caller func?